### PR TITLE
Add Linux Transport benchmark

### PR DESCRIPTION
**Pull request by tmds**
Per https://github.com/aspnet/benchmarks/issues/255

@davidfowl

### CODE REVIEW

File: build/repo.targets [hunk 1]
```diff
@@ -1,9 +1,11 @@
 <Project>
   <ItemGroup>
     <!-- Optimal ThreadCount for current hardware -->
-    <Scenarios Include="-n Plaintext --kestrelThreadCount 2" />
-    
+    <Scenarios Include="-n Plaintext --kestrelTransport Libuv --kestrelThreadCount 2" />
```
Comment (tmds) [inline build/repo.targets]: Before setting kestrelThreadCount implied libuv. That is no longer the case. Made libuv explicit here.

File: build/repo.targets [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
+    <Scenarios Include="-n Plaintext --kestrelTransport RHLinux --kestrelThreadCount 2" />
```
Comment (tmds) [inline build/repo.targets]: I added one scenario with kestrelThreadCount and one without.
The one with is for the threadpool dispatch, the one without is for the inline dispatch.

File: src/Benchmarks/Program.cs [hunk 3]
```diff
@@ -75,13 +77,13 @@ public static void Main(string[] args)
                         // Dispatching to the thread pool means we don't want to use the transport thread
                         options.UseTransportThread = !bool.Parse(kestrelThreadPoolDispatchingValue);
                     }
+                    threadPoolDispatching = !options.UseTransportThread;
```
Comment (tmds) [inline src/Benchmarks/Program.cs]: threadPoolDispatching was not used. So I assigned it here.

File: src/Benchmarks/Program.cs [hunk 4]
```diff
@@ -97,10 +99,28 @@ public static void Main(string[] args)
                         }
                     });
                 }
-                else if (string.Equals(kestrelTransport, "Sockets", StringComparison.OrdinalIgnoreCase))
+                else if (string.Equals(kestrelTransport, "Sockets", StringComparison.OrdinalIgnoreCase)
+                         && threadPoolDispatching != true)
```
Comment (tmds) [inline src/Benchmarks/Program.cs]: This avoids benchmarking the double dispatch.

File: src/Benchmarks/Program.cs [hunk 5]
```diff
@@ -97,10 +99,28 @@ public static void Main(string[] args)
...
                 {
                     webHostBuilder.UseSockets();
                 }
+                else if (string.Equals(kestrelTransport, "RHLinux", StringComparison.OrdinalIgnoreCase))
+                {
+                    if (!RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                    {
+                        throw new PlatformNotSupportedException("RHLinux Transport requires Linux");
+                    }
+#if NETCOREAPP2_0
```
Comment (davidfowl) [inline src/Benchmarks/Program.cs]: Why does this require netcoreapp2.0?

Comment (tmds) [inline src/Benchmarks/Program.cs]: The app targets `<TargetFrameworks>netcoreapp2.0;net46</TargetFrameworks>`.
And the library targets `netstandard2.0`.
The netstandard2.0 library doesn't work on the net46 target (net461 needed for netstandard2.0).
I think I can make this work on netstandard1.3. Then this condition (+ matching one in csproj) can go.

**Comment by tmds** [comment]
hi @davidfowl I still like to add this to the benchmark repo. I believe this wasn't merged due to a dependency on infrastructure changes (https://github.com/aspnet/Performance/issues/226). I see that issue still isn't resolved. Perhaps we are no longer depending on it? Let me know when I can bring this PR up to date so it can be merged.

**Comment by sebastienros** [comment]
Done on TE

**Pull request closed without merging**